### PR TITLE
Relaunch: Homepage FxA Experiment (Growth Exp 2)

### PR DESCRIPTION
**Pull request by maxxcrawford**
### Treatment

**Note: When testing locally, add the variable `EXPERIMENT_ACTIVE=1` to your `.ENV` file.**

1. Go to https://monitor-v2.herokuapp.com/?experimentBranch=vb
1. Click "Get Started" Button
1. Go through FxA Flow 
1. Once completed, user should be on the `/user/dashboard/` page

## Expected Results

**Control:** 
image

**Treatment:** 
_Note - The text is still being worked on._

image

### CODE REVIEW

File: template-helpers/homepage.js [hunk 1]
```diff
@@ -28,6 +32,11 @@ function makeLanding(args) {
     });
   }
 
+  // Growth Experiment
+  if (featuredBreach && experimentBranchB) {
+    landingCopy.subhead = "Your free Firefox account alerts you if you’ve been involved in a known data breach.";
```
Comment (lesleyjanenorton) [inline template-helpers/homepage.js:37]: Just verifying - do we want this to be English only?

File: controllers/user.js [hunk 2]
```diff
@@ -230,6 +229,18 @@ async function getDashboard(req, res) {
   const allBreaches = req.app.locals.breaches;
   const { verifiedEmails, unverifiedEmails } = await getAllEmailsAndBreaches(user, allBreaches);
 
+  let experimentBranch = null;
+  let isUserInExperiment = null;
+  let experimentBranchB = null;
```
Comment (groovecoder) [inline controllers/user.js:234]: `let` = code sniff. Can we wrap these up in a single object and do something more like:
```js
const experimentFlags = _getExperimentFlags(req);
...
res.render("dashboards", {
...
  experimentFlags,
});
```

File: controllers/home.js [hunk 3]
```diff
@@ -15,10 +17,24 @@ async function home(req, res) {
   let featuredBreach = null;
   let scanFeaturedBreach = false;
 
+  // Growth Experiment
+  let experimentBranch = null;
+  let isUserInExperiment = null;
+  let experimentBranchB = null;
```
Comment (groovecoder) [inline controllers/home.js:23]: `let` = code sniff. Can we wrap these up in a single object and do something more like:
```js
const experimentFlags = _getExperimentFlags(req);
...
res.render("dashboards", {
...
  experimentFlags,
});
```

File: scan-results.js [hunk 4]
```diff
@@ -6,28 +6,28 @@ const HIBP = require("./hibp");
...
   const allBreaches = req.app.locals.breaches;
   let scannedEmail = null;
 
   // Growth Experiment
-  // let experimentBranch = null;
-  // let isUserInExperiment = null;
-  // let experimentBranchB = null;
+  let experimentBranch = null;
+  let isUserInExperiment = null;
+  let experimentBranchB = null;
```
Comment (groovecoder) [inline scan-results.js:20]: `let` = code sniff. Can we wrap these up in a single object and do something more like:
```js
const experimentFlags = _getExperimentFlags(req);
...
res.render("dashboards", {
...
  experimentFlags,
});
```

File: template-helpers/homepage.js [hunk 5]
```diff
@@ -5,6 +5,7 @@ const { LocaleUtils } = require("./../locale-utils");
 function makeLanding(args) {
   const locales = args.data.root.req.supportedLocales;
   const featuredBreach = args.data.root.featuredBreach;
+  const experimentBranchB = args.data.root.experimentBranchB;
```
Comment (groovecoder) [inline template-helpers/homepage.js:8]: If we change the controllers to add an `experimentFlags` object to the template context, we'll need to update these references to `args.data.root.experimentFlags.experimentBranchB`, etc.

File: views/monitor.hbs [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {{!< default }}
-<main class="container monitor-homepage clear-header {{#if featuredBreach}} featured-breach {{/if}} {{#if featuredBreach.IsSensitive}} sensitive-landing {{/if}} data-page-label={{> analytics/landing_page_label }}>
+<main class="container monitor-homepage clear-header {{#if featuredBreach}} featured-breach {{/if}} {{#if featuredBreach.IsSensitive}} sensitive-landing {{/if}} {{#if experimentBranchB}} experiment-subhead {{/if}}" data-page-label={{> analytics/landing_page_label }}>
```
Comment (groovecoder) [inline views/monitor.hbs:2]: Might need to be `experimentFlags.experimentBranchB`

File: template-helpers/homepage.js (hunk 1 above)
Comment (maxxcrawford) [reply template-helpers/homepage.js:37]: @lesleyjanenorton @groovecoder Correct – it's being filtered in this argument: 
https://github.com/mozilla/blurts-server/blob/1638-homepage-fxa-experiment/controllers/home.js#L31

We're using a direct string like this in order to not deploy any l10n resources.

**Comment by maxxcrawford** [comment]
> Can we try to remove the `let` assignments and introduce a `getExperimentFlags` function instead?
> 
> If that becomes too much work, we can merge without the refactor.

Migrated the fix into another issue (#1698) and PR (https://github.com/mozilla/blurts-server/pull/1699)

**Pull request merged**